From 3df970e7bd2ddb53af1d3c8f158f45d4c4cbb289 Mon Sep 17 00:00:00 2001
From: Yuta Saito <kateinoigakukun@gmail.com>
Date: Wed, 14 Feb 2024 12:43:15 +0000
Subject: [PATCH] HACK: [wasm] Hard-define wasi-libc constants

They are defined in the way ClangImporter can't handle them
---
 Sources/Foundation/FileManager+POSIX.swift | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/Sources/Foundation/FileManager+POSIX.swift b/Sources/Foundation/FileManager+POSIX.swift
index 2150bd11..9b877dbc 100644
--- a/Sources/Foundation/FileManager+POSIX.swift
+++ b/Sources/Foundation/FileManager+POSIX.swift
@@ -15,6 +15,13 @@ internal func &(left: UInt32, right: mode_t) -> mode_t {
 
 #if os(WASI)
 import WASILibc
+let DT_DIR: UInt8 = 0x3
+let O_WRONLY: Int32 = 0x10000000
+let O_CREAT: Int32 = 1 << 12
+let O_DIRECTORY: Int32 = 1 << 13
+let O_EXCL: Int32 = 1 << 14
+let O_TRUNC: Int32 = 1 << 15
+
 internal var errno : Int32 {
   get {
     return _swift_stdlib_getErrno()
-- 
2.43.0

